src/cairo/src/utils.cairo

```diff
@@ -0,0 +1,74 @@
+use array::ArrayTrait;
+const STARK_MINUS_1: u256 = 3618502788666131213697322783095070105623107215331596699973092056135872020480;
+
+// From a 128 bit scalar, decomposes it into base (-3) such that
+// scalar = sum(digits[i] * (-3)^i for i in [0, 81])
+// scalar = sum_p - sum_n
+// Where sum_p = sum(digits[i] * (-3)^i for i in [0, 81] if digits[i]==1)
+// And sum_n = sum(digits[i] * (-3)^i for i in [0, 81] if digits[i]==-1)
+// Returns (abs(sum_p), abs(sum_n), p_sign, n_sign)
+pub fn  scalar_to_base_neg3_le(scalar: u128) -> (felt252, felt252, felt252, felt252){
+
+    let mut scalar: u128 = scalar;
+
+    let digits: Span<felt252> = neg_3_base_le(scalar).span();
+    let digits_len = digits.len();
+
+
+    let mut sum_p = 0;
+    let mut sum_n = 0;
+
+    let mut base_power = 1;
+    let mut i = 0;
+
+    //implementation like this to run in O(n) time complexity, only computing each power of -3 once
+    while i < digits_len {
```

Comment:
Try to find a way such that the exit condition of the loop is not a comparison but rather a !=0 check. 
This is the most performant.

Reply:
Did it now, but doesn't seem to make much difference